### ISSUE TITLE
Error setting up

### ISSUE THREAD

**Issue by Cryonic90** [MTrab/landroid_cloud#175]
I have a problem since version 2.1.0 that setting up the integration doesn't work anymore.
It fails with the following error:
```
Error setting up entry Worx - cryonic90@gmail.com for landroid_cloud
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 339, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/landroid_cloud/__init__.py", line 61, in async_setup_entry
    result = await _setup(hass, entry)
  File "/config/custom_components/landroid_cloud/__init__.py", line 203, in _setup
    await hass.async_add_executor_job(cloud.connect, False, False)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyworxcloud/__init__.py", line 224, in connect
    self._fetch()
  File "/usr/local/lib/python3.9/site-packages/pyworxcloud/__init__.py", line 609, in _fetch
    device = DeviceHandler(self._api, product)
  File "/usr/local/lib/python3.9/site-packages/pyworxcloud/utils/devices.py", line 36, in __init__
    self.__mapinfo(api, product)
  File "/usr/local/lib/python3.9/site-packages/pyworxcloud/utils/devices.py", line 84, in __mapinfo
    self.warranty = Warranty(data)
  File "/usr/local/lib/python3.9/site-packages/pyworxcloud/utils/warranty.py", line 25, in __init__
    self["expires_at"]
TypeError: '<' not supported between instances of 'NoneType' and 'datetime.datetime'
```
What does this mean?

**Comment by MTrab** [comment]
This is a duplicate of https://github.com/MTrab/pyworxcloud/issues/44

Ideally this wouldn't cast any errors, it basically means that you haven't activated your warranty for the device.

A fix is underway in PR #173

**Comment by Cryonic90** [comment]
Ah yes, sorry, didn't see that one.
But adding the warranty information did fix it for me too.
Before that it didn't complete the setup :)
